tests/e2e/ambient/ambient_test.go

```diff
@@ -0,0 +1,365 @@
+							return err
+						}).Should(Succeed(), "Expected to retrieve the 'istiod' deployment")
+
+						Expect(istiodObj).To(HaveContainersThat(ContainElement(WithTransform(getEnvVars,
+							ContainElement(corev1.EnvVar{Name: "PILOT_ENABLE_AMBIENT", Value: "true"})))),
+							"Expected PILOT_ENABLE_AMBIENT to be set to true, but not found")
+
+						Expect(istiodObj).To(HaveContainersThat(ContainElement(WithTransform(getEnvVars,
+							ContainElement(corev1.EnvVar{Name: "CA_TRUSTED_NODE_ACCOUNTS", Value: "ztunnel/ztunnel"})))),
+							"Expected CA_TRUSTED_NODE_ACCOUNTS to be set to ztunnel/ztunnel, but not found")
+					})
+				})
+
+				When("the ZTunnel CR is created", func() {
+					BeforeAll(func() {
+						ztunnelYaml := `
+apiVersion: sailoperator.io/v1alpha1
+kind: ZTunnel
+metadata:
+  name: default
+spec:
+  profile: ambient
+  version: %s
+  namespace: %s`
+						ztunnelYaml = fmt.Sprintf(ztunnelYaml, version.Name, ztunnelNamespace)
+						Log("ZTunnel YAML:", ztunnelYaml)
+						Expect(k.CreateFromString(ztunnelYaml)).To(Succeed(), "ZTunnel creation failed")
+						Success("ZTunnel created")
+					})
+
+					It("deploys the ZTunnel DaemonSet", func(ctx SpecContext) {
+						Eventually(func(g Gomega) {
+							daemonset := &appsv1.DaemonSet{}
+							g.Expect(cl.Get(ctx, kube.Key("ztunnel", ztunnelNamespace), daemonset)).To(Succeed(), "Error getting ZTunnel DaemonSet")
+							g.Expect(daemonset.Status.NumberAvailable).
+								To(Equal(daemonset.Status.CurrentNumberScheduled),
+									"ZTunnel DaemonSet Pods not Available; expected numberAvailable to be equal to currentNumberScheduled")
+						}).Should(Succeed(), "ZTunnel DaemonSet Pods are not Available")
+						Success("ZTunnel DaemonSet is deployed and Running")
+					})
+
+					It("has ztunnel running with appropriate env variables set", func(ctx SpecContext) {
+						var ztunnelObj appsv1.DaemonSet
+
+						Eventually(func() error {
+							_, err := common.GetObject(ctx, cl, kube.Key("ztunnel", ztunnelNamespace), &ztunnelObj)
+							return err
+						}).Should(Succeed(), "Expected to retrieve the 'ztunnel' daemonSet")
+
+						Expect(ztunnelObj).To(HaveContainersThat(ContainElement(WithTransform(getEnvVars,
+							ContainElement(corev1.EnvVar{Name: "XDS_ADDRESS", Value: "istiod.istio-system.svc:15012"})))),
+							"Expected XDS_ADDRESS to be set to istiod.istio-system.svc:15012, but not found")
+
+						Expect(ztunnelObj).To(HaveContainersThat(ContainElement(WithTransform(getEnvVars,
+							ContainElement(corev1.EnvVar{Name: "ISTIO_META_ENABLE_HBONE", Value: "true"})))),
+							"Expected ISTIO_META_ENABLE_HBONE to be set to true, but not found")
+					})
+				})
+
+				// We spawn the following pods to verify the data-path connectivity.
+				// an httpbin service in httpbin namespace that listens of port 8000
+				// using a sleep pod from the sleep namespace, we try to connect to the httpbin service to verify that connectivity is successful.
```

Comment:
What does this test validate ? A sleep service can still access a httpbin service even without ambient mode or label , right ? 
I think we should at least validate a ztunnel log about the sleep, httpbin connectivity log. 
For example, `kubectl -n ztunnel logs -l app=ztunnel | grep -E "inbound|outbound"`

Reply:
Rather than looking at the logs, lets verify that the ztunnel has opened the listening sockets in the pod network namespace (more specifically there is a socket listening on HBONE port). Let me update the PR.